### PR TITLE
 feature/zainnk902/class-recurring-events

### PR DESCRIPTION
**Pull request by zainnk902**
### Administrative Info

Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Github Issue]-[3-4 word description separated by dashes]`.

### What issue(s) does this branch close?
Refactors how the date and time is displayed for classes, also accommodates recurring events using rrule library
### Changes

What changes did you make?

- I changed the Class object to take in an rrstring, that would represent the rrule object
- Changed the ClassCard component to make an rrule object from the rrstring, and format the dates and times accordingly
- Converted the starting and ending time for classes to ISO time

### Testing

How did you confirm your changes worked?

- Took in dummy class objects in "league.ts" to see if the output would still display the days of the week and times the classes are, and it was consistent.

### Confirmation of Change
Screen Shot 2022-02-22 at 2 10 46 PM

### CODE REVIEW

File: components/ClassCard.tsx [hunk 1]
```diff
@@ -14,25 +15,25 @@ const ClassCard: React.FC<ClassCardProps> = ({
...
+  //geting all days of week from the rrule object to output
+  const dates = tempRule.substring(tempRule.indexOf("on") + 3, tempRule.indexOf("until"));
+  //this takes in teh start and end times, converts them to ISO format, then outputs the hour the class starts and ends
   const convertTime = (startTime: string, endTime: string) => {
-    const startTimeStr: string = moment(moment().format(startTime), "HH:mm").format("h");
-    const endTimeStr: string = moment(moment().format(endTime), "HH:mm").format("h A");
-    return startTimeStr + "-" + endTimeStr;
+    const { DateTime } = require("luxon");
+    const startTime_ISO = DateTime.fromISO(startTime);
+    const endTime_ISO = DateTime.fromISO(endTime);
```
Comment (Anshul-Birla) [inline components/ClassCard.tsx:30]: this unfortunately doesn't do what wre need it to do. We need to convert from the given ISO string, to the local time. So first, we need do `DateTime.fromISO(startTime).toLocal().toFormat('h')` to get the correctly formatted start time. You can construct `finalTime` with that

File: package.json [hunk 2]
```diff
@@ -20,13 +20,15 @@
     "http-status-codes": "^2.2.0",
     "io-ts": "^2.2.16",
     "io-ts-promise": "^2.0.2",
+    "luxon": "^2.3.0",
     "moment": "^2.29.1",
```
Comment (Anshul-Birla) [inline package.json:24]: can we delete this. I don't think we are using moment anymore

File: components/ClassCard.tsx [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import moment from "moment";
 import styles from "../styles/components/LeagueViews.module.css";
+import { RRule, RRuleSet, rrulestr } from "rrule";
```
Comment (Anshul-Birla) [inline components/ClassCard.tsx:3]: lets remove all of the unnecessary imports after your clean the code up

File: components/ClassView.tsx [hunk 4]
```diff
@@ -31,7 +31,7 @@ const ClassView: React.FC<ClassViewProp> = ({ classes }) => (
               name={c.name}
```
Comment (Anshul-Birla) [inline components/ClassView.tsx:31]: instead of using `c`, can we use something more informative like `class`

File: pages/league.tsx [hunk 5]
```diff
@@ -38,10 +39,10 @@ const League: NextPage = () => {
     name: "Intro to Java",
     minLevel: 3,
     maxLevel: 5,
-    recurrence: [1, 2, 3],
+    rrstring:
+      "DTSTART:20220222T093000Z\nRRULE:FREQ=WEEKLY;UNTIL=20230222T093000Z;BYDAY=SU,MO,WE,FR;INTERVAL=1",
     timeStart: "13:00",
     timeEnd: "14:00",
```
Comment (Anshul-Birla) [inline pages/league.tsx:45]: lets use actual ISO strings. Maybe something like `07:34Z` and `08:34Z`

**Review by Anshul-Birla — CHANGES_REQUESTED** [review]
Some things to change. 
- Install the luxon types. Should just be the command listed [here](https://www.npmjs.com/package/@types/luxon)
- Use ISO 8601 time strings in the dummy data. The Luxon documentation has an example under the `toISOTime()`
- Update your node to 16.13.2. Use [nvm](https://github.com/nvm-sh/nvm) if you need to. After that is done, remove your package-lock.json and node modules, and reinstall everything. Your package.json should have `lockfile-version: 2` in it.

**Review by Anshul-Birla — APPROVED** [review]
LGTM :)

**Review by nishantbalaji — APPROVED** [review]
lgtm 😄

**Review by Anshul-Birla — APPROVED** [review]
Done :)!

**Review by nishantbalaji — APPROVED** [review]
lgtm

**Pull request merged**